src/Prelude.mo

```diff
@@ -0,0 +1,25 @@
+module {
+  public module Option {
+
+    public func guard(condition : Bool) : ?() = do ? {
+      if (not condition) null!
+    };
+  };
+
+  public module Num {
+
+    public type Mod = Nat;
+
+    type Mod_ = {
+      mod : Nat -> ?Mod;
+      add : (Mod, Nat) -> ?Mod;
+      sub : (Mod, Nat) -> ?Mod;
+    };
+
+    public func Mod(n : Nat) : Mod_ = object {
+      public func mod(x : Nat) : ?Mod = if (n == 0) null else ?(x % n);
+      public func add(idx : Mod, x : Nat) : ?Mod = mod(idx + x);
+      public func sub(idx : Mod, x : Nat) : ?Mod = do ? { add(idx, n - mod(x)!)! };
```

Comment:
It chains undesired behavior as `null`s, following to the principle of correctness by construction: eliminating incorrect data representation, not validating it.

Reply:
It just a nice sugar for those chains and maps, with no necessity to specify each callback type parameter (according to the Motoko's limited type inference abilities, and also because of a bit verbose lambda syntax).

Reply:
Otherwise it would be necessary to do nested switching with repetitive `return null`.